src/query/storage/converter.go

```diff
@@ -260,28 +260,32 @@ func iteratorToTsSeries(
 	iter encoding.SeriesIterator,
 	enforcer cost.ChainedEnforcer,
 	tagOptions models.TagOptions,
-) (*ts.Series, error) {
+) (*ts.Series, ts.SeriesExemplar, error) {
 	metric, err := FromM3IdentToMetric(iter.ID(), iter.Tags(), tagOptions)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	datapoints := make(ts.Datapoints, 0, initRawFetchAllocSize)
+	exemplars := make(ts.SeriesExemplar, 0)
 	for iter.Next() {
-		dp, _, _ := iter.Current()
+		dp, _, annotation := iter.Current()
 		datapoints = append(datapoints, ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value})
+		if len(annotation) > 0 {
```

Comment:
Add exemplars only when they were requested, otherwise avoid sending them down the line.